pandas/io/parsers/readers.py

```diff
@@ -505,11 +505,11 @@ def read_csv(
     delimiter=None,
     # Column and Index Locations and Names
     header="infer",
-    names=lib.no_default,
+    names=None,
```

Comment:
i would rather leave these as no_default and fix the check itself

Reply:
We should only have one way of representing none/null, so we should either use ``None`` or ``lib.no_default``, not have both. I think I prefer ``None`` given that ``lib.no_default`` is private. 

If we want to keep using ``lib.no_default`` then, we should probably close OP as wontfix.

Reply:
no i think you can simply revert these back to lib.no_default and fix the check on L1305. the point is that these are *still* invalid if actual `None` is passed.

Reply:
Not sure iiuc, but ``pd.read_csv(xxx, names=None, prefix=None)`` should fail?

Reply:
I think it should work. It worked at least in previous versions. If ``None`` is passed for both, it would just get the column names from header, right?

Reply:
right i agree this should work, but i also think you can leave the no_default and it will work (just odify the comparison)

Reply:
OK, thx for clarifying, update incoming soon.